pkg/execution/state/state.go

```diff
@@ -95,15 +97,27 @@ type Loader interface {
 // a map[string]interface{} containing event data.
 type Mutater interface {
 	// New creates a new state for the given run ID, using the event as the input data for the root workflow.
-	New(ctx context.Context, workflow inngest.Workflow, runID ulid.ULID, input any) (State, error)
+	New(ctx context.Context, workflow inngest.Workflow, runID ulid.ULID, input map[string]any) (State, error)
 
 	// SaveActionOutput stores output for a single action within a workflow run.
+	//
+	// This should clear any error that exists for the current action, indicating that
+	// the step is a success.
 	SaveActionOutput(ctx context.Context, i Identifier, actionID string, data map[string]interface{}) (State, error)
 
 	// SaveActionError stores an error for a single action within a workflow run.  This is
 	// considered final, as in the action will not be retried.
```

Comment:
This line saying that it's considered final and the action won't be retried seems like it is contrary to the other comments about overwriting the error on a successful action or passing `nil` to this method. Maybe its just a comment we need to update?